builds/e2e/e2e.yaml

```diff
@@ -285,6 +285,64 @@ jobs:
       parameters:
         sas_uri: $(sas_uri)
 
+################################################################################
+  - job: ubuntu_2404
+################################################################################
+    displayName: Ubuntu 24.04 on amd64
+    dependsOn: Token
+    condition: succeeded('Token')
+
+    pool:
+      name: $(pool.custom.name)
+      demands: ubuntu2404-amd64-e2e-tests
```

Comment:
Why are we using custom pool and not the `$(pool.linux.name)`?

Reply:
Unfortunately, 1ES hosted pools do not yet support Ubuntu 24.04. So we'll need to use a custom agent until they do.